File: android-native/app/src/main/java/com/example/kpkn/data/repository/AugeMetricsRepository.kt
package com.example.kpkn.data.repository

import com.example.kpkn.data.models.ExerciseMuscleInfo

/**
 * Métricas AUGE extraídas directamente de la base de datos de ejercicios.
 * Nunca contiene valores inventados: si la DB no tiene el campo, el campo
 * correspondiente es null y el caller debe mostrar "—" o saltar el cálculo.
 *
 * @param efc  Costo metabólico/muscular (1–5). Null si no declarado en DB.
 * @param cnc  Costo del Sistema Nervioso Central (1–5). Null si no declarado.
 * @param ssc  Costo espinal/estructural (0–2). Null si no declarado.
 * @param axialLoad Factor de carga axial (0.0–1.0). Null si no declarado o no aplica.
 */
data class AugeDbMetrics(
    val efc: Double,
    val cnc: Double,
    val ssc: Double,
    val axialLoad: Double?,
) {
    /** SNC (Sistema Nervioso Central) — alias semántico de cnc (central nervous cost). */
    val snc: Double get() = cnc
}

/**
 * Repositorio de métricas AUGE. Única API para leer EFC/CNC/SSC en runtime.
 * Retorna null si la DB no tiene datos completos — nunca inventa valores.
 *
 * Uso:
 * ```kotlin
 * val repo = AugeMetricsRepository(exerciseDb.associateBy { it.id.lowercase() })
 * val metrics = repo.metricsFor(exercise.exerciseDbId)
 *     ?: return  // ejercicio sin datos: saltar o mostrar "—"
 * ```
 */
class AugeMetricsRepository(
    private val exerciseIndex: Map<String, ExerciseMuscleInfo>,
) {
    /**
     * Retorna métricas AUGE para el ejercicio dado.
     *
     * @param exerciseDbId ID del ejercicio en la DB (puede ser null si el ejercicio es personalizado).
     * @return [AugeDbMetrics] si la DB tiene efc+cnc+ssc completos, null en caso contrario.
     */
    fun metricsFor(exerciseDbId: String?): AugeDbMetrics? {
        val id = exerciseDbId?.lowercase() ?: return null
        val info = exerciseIndex[id] ?: return null
        val efc = info.efc ?: return null
        val cnc = info.cnc ?: return null
        val ssc = info.ssc ?: return null
        return AugeDbMetrics(
            efc = efc.coerceIn(1.0, 5.0),
            cnc = cnc.coerceIn(1.0, 5.0),
            ssc = ssc.coerceIn(0.0, 2.0),
            axialLoad = info.axialLoadFactor,
        )
    }

    companion object {
        /** Crea un repositorio a partir de la lista de ejercicios de la DB. */
        fun from(exerciseList: List<ExerciseMuscleInfo>): AugeMetricsRepository =
            AugeMetricsRepository(exerciseList.associateBy { it.id.lowercase() })
    }
}

File: ios-native/KPKNFit/KPKNFit/Data/Repository/AugeMetricsRepository.swift
import Foundation

public struct AugeDbMetrics: Codable {
    public let efc: Double
    public let cnc: Double
    public let ssc: Double
    public let axialLoad: Double?

    public init(efc: Double, cnc: Double, ssc: Double, axialLoad: Double? = nil) {
        self.efc = efc
        self.cnc = cnc
        self.ssc = ssc
        self.axialLoad = axialLoad
    }

    public var snc: Double { cnc }
}

public final class AugeMetricsRepository {
    private let exerciseIndex: [String: ExerciseMuscleInfo]

    public init(exerciseIndex: [String: ExerciseMuscleInfo] = [:]) {
        self.exerciseIndex = exerciseIndex
    }

    public func metricsFor(exerciseDbId: String?) -> AugeDbMetrics? {
        guard let id = exerciseDbId?.lowercased() else { return nil }
        guard let info = exerciseIndex[id] else { return nil }
        guard let efc = info.efc else { return nil }
        guard let cnc = info.cnc else { return nil }
        guard let ssc = info.ssc else { return nil }
        return AugeDbMetrics(
            efc: min(max(efc, 1.0), 5.0),
            cnc: min(max(cnc, 1.0), 5.0),
            ssc: min(max(ssc, 0.0), 2.0),
            axialLoad: info.axialLoadFactor
        )
    }

    public static func from(exerciseList: [ExerciseMuscleInfo]) -> AugeMetricsRepository {
        let index = Dictionary(exerciseList.map { ($0.id.lowercased(), $0) }, uniquingKeysWith: { first, _ in first })
        return AugeMetricsRepository(exerciseIndex: index)
    }
}
